### ISSUE TITLE
Probably wrong error reporting for "no-call-expression"

### ISSUE THREAD

**Issue by svzi** [angular-eslint/angular-eslint#374]
Hi there!

First of all thank you very much for creating and maintaining this awesome plugin, I really appreciate it! 👏 

We used TSLint in our Angular project before and recently made the switch to ESLint. We nearly managed to get everything going like before, but this rule lets us scratching our heads:
`@angular-eslint/template/no-call-expression` (TSLint: `template-no-call-expression`)

Afaik this rule is supposed to prevent coding like this in the template:
```html
<div>{{ getTitle() }}</dv>
```
That's a bad practice because the function will be called on every change detection cycle. The TSLint implementation worked like a charm for us. Sadly the ESLint implementation doesn't. Because it will also report this piece of code as faulty:
```html
<div (click)="doSomething()">...
```
I don't think the linter should mark this code as an error, because this method won't be executed in every change detection cycle. It will only get called when the `click` event occurs.

Because I didn't find a way to configure this rule I was forced to deactivate it for our whole project. But we would love to  reactivate it, because it helped us a lot in the past.

Just in case this rule should be a replacement for the already mentioned TSLint rule, then I think here is something going wrong. But maybe I'm just misunderstanding the rule. Any feedback would be really much appreciated! 🙏 

Best,
Sven

**Comment by JamesHenry** [comment]
Hi @svzi, we have an explicit unit test for the rule which asserts that there should be no error for event bindings: https://github.com/angular-eslint/angular-eslint/blob/d7c6aa49132100de23bd101f36ffccf6a8b6414e/packages/eslint-plugin-template/tests/rules/no-call-expression.test.ts#L24

This would suggest that there is something more complex going on. Please could you therefore create a full reproduction repo with the issue in for us to investigate.

Many thanks!

**Comment by JamesHenry** [comment]
Please feel free to reopen this whenever you have chance to reproduce @svzi but as mentioned our current understanding is that this case is covered, so closing for now

**Comment by hisomu** [comment]
I am also seeing a similar issues, however it's around forms

`<form [formGroup]="form" (ngSubmit)="form.valid && save()">`

**Comment by rafaelss95** [comment]
> I am also seeing a similar issues, however it's around forms
> 
> `<form [formGroup]="form" (ngSubmit)="form.valid && save()">`

I was able to reproduce not only this case, but also with `||` and ternary. I submitted #390 that aims to fix this behavior.
